js/weather.js

```diff
@@ -1,35 +1,91 @@
 $(function () {
+  let selectedBox = 'current';
+  let apiData = {};
+
+  $('.toggle-buttons').click(function (e) {
+    // prevent checked box to be unchecked on click
+    if (!$(this).is(':checked')) {
+      e.preventDefault;
+      return false;
+    }
+    $(this).siblings().attr('checked', false); // uncheck checked siblings
+    selectedBox = $(this)[0].value;
+
+    if (Object.keys(apiData).length) {
+      displayResults();
+    }
+  });
+
   // React to hitting enter in the text box instead of clicking submit.
-  $("#search-weather").submit(function(e) {
+  $('#search-weather').submit(function (e) {
     e.preventDefault();
     clearResult('');
 
     // Get the weather info for the selected search location.
-    $.get("/weather.php?query=" + $("#search").val(), function (data) {
+    $.get('/weather.php?query=' + $('#search').val(), function (data) {
+      apiData = data;
+      console.log(data);
       // Show weather results.
       if (data.temp_f) {
-        showResult(data);
+        prepResult(apiData);
       } else {
         clearResult('No valid data for your location.');
       }
     });
   });
 
   function clearResult($msg = 'Sorry Charlie! Not a valid input.') {
-    $(".weather_icon").attr("src", "images/trans.png");
-    $("#location").hide();
-    $("#city").text();
-    $("#state").text();
-    $(".result .description").html($msg);
+    $('.weather_icon').attr('src', 'images/trans.png');
+    $('.result').hide();
+    $('#city').text();
+    $('#state').text();
+    $('.forecast').html('');
+    $('.result .description').html($msg);
+    apiData = {};
   }
 
-  function showResult(data) {
-    $(".weather_icon").attr("src", data.icon_url);
-    $("#location").show();
-    $("#city").text(data.city);
-    $("#state").text(data.state);
-    var desc = data.weather + " and " + data.temp_f + "&deg; F";
-    $(".result .description").html(desc);
+  function prepResult(data) {
```

Comment:
I like this separation of steps into functions that each have a focused responsibility (prep vs. display).